src/benchmarklib/benchmark_table_encoder.cpp

```diff
@@ -0,0 +1,150 @@
+#include "benchmark_table_encoder.hpp"
+
+#include "constant_mappings.hpp"
+#include "encoding_config.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_encoded_segment.hpp"
+#include "storage/base_value_segment.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseValueSegment&) { return {EncodingType::Unencoded}; }
+
+SegmentEncodingSpec generate_segment_encoding_spec(const ReferenceSegment&) {
+  Fail("Did not expect a ReferenceSegment in base table");
+}
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseEncodedSegment& base_encoded_segment) {
+  auto vector_compression_type = std::optional<VectorCompressionType>{};
+
+  switch (base_encoded_segment.compressed_vector_type()) {
+    case CompressedVectorType::Invalid:
+      vector_compression_type = VectorCompressionType::FixedSizeByteAligned;
```

Comment:
Why is this not a fail case?

Reply:
You know what? There is no such thing as "CompressedVectorType::Invalid", I'll purge it from hyrise.